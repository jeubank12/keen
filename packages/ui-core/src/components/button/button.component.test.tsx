import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Button from './button.component';

describe('<Button />', () => {
  const children = <span>Sign up</span>;

  it('should render children', () => {
    const wrapper = mount(<Button>{children}</Button>);

    expect(wrapper.contains(children)).toBeTruthy();
  });

  it('should render "anchor" element', () => {
    const wrapper = mount(<Button href="https://keen.io">{children}</Button>);

    expect(wrapper.find('a').length).toBeTruthy();
  });

  it('should call "onClick" handler', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<Button onClick={mockFn}>{children}</Button>);
    wrapper.simulate('click');

    expect(mockFn).toHaveBeenCalled();
  });

  it('should not call "onClick" handler for disable state', () => {
    const mockFn = jest.fn();
    const wrapper = mount(
      <Button isDisabled={true} onClick={mockFn}>
        {children}
      </Button>
    );
    wrapper.simulate('click');

    expect(mockFn).not.toHaveBeenCalled();
  });

  it('should apply styles for button with type "primary"', () => {
    const wrapper = mount(<Button type="primary">{children}</Button>);

    expect(wrapper).toMatchSnapshot();
  });
});
