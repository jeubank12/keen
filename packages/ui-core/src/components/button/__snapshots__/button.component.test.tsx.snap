// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/ui-core - <Button /> should apply styles for button with "primary" variant 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 15px;
  font-family: 'Lato Bold',sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 22.5px;
  padding: 0 25px;
  outline: none;
  border: none;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  background-color: #F0AB30;
  border-color: #F0AB30;
  color: #1D2729;
  height: 42px;
}

.c0:focus {
  box-shadow: 0 0 0 0.2rem rgba(218,165,80,.5);
}

.c0:hover {
  color: #273437;
  background-color: #E29B1E;
}

<Component
  variant="primary"
>
  <styled.button
    isDisabled={false}
    onClick={[Function]}
    size="default"
    type="button"
    variant="primary"
  >
    <button
      className="c0"
      onClick={[Function]}
      size="default"
      type="button"
    >
      <span>
        Sign up
      </span>
    </button>
  </styled.button>
</Component>
`;
